<b>Importing the Required Libraries

In [1]:
import pandas as pd

<b>Reading the Data

In [2]:
data = pd.read_csv('mobile_events_2020.csv',sep=";") #CSV file is separated by semicolons
data.head() #Reading the Top 5 Records

anonymous_id business_zoned context_app_version  \
0  9DDC5BCD-0477-4E83-A32A-51013DA0A4E7        Hamburg              3.1.13   
1  5592E78E-CE87-4258-B8D7-DD834A687A1E        Hamburg              3.1.13   
2  5592E78E-CE87-4258-B8D7-DD834A687A1E        Hamburg              3.1.13   
3  5592E78E-CE87-4258-B8D7-DD834A687A1E        Hamburg              3.1.13   
4  5592E78E-CE87-4258-B8D7-DD834A687A1E        Hamburg              3.1.13   

                      context_device_id context_os_name  \
0  7A718FF6-584F-40E2-939D-224231295B1E             iOS   
1  11D0CB16-DA38-4565-9C3B-066E7877856C             iOS   
2  11D0CB16-DA38-4565-9C3B-066E7877856C             iOS   
3  11D0CB16-DA38-4565-9C3B-066E7877856C             iOS   
4  11D0CB16-DA38-4565-9C3B-066E7877856C             iOS   

                        event_name          received_at           created_at  
0  Out-Of-Business Zone View Shown  2020-09-01 23:59:44  2020-09-01 23:59:18  
1               Application Opened  2020-09-01 23:59:41  2020-09-01 23:59:35  
2           Ride Done - Successful  2020-09-01 23:59:41  2020-09-01 23:59:37  
3               Application Opened  2020-09-01 23:59:36  2020-09-01 23:37:46  
4         Ride Done - Unsuccessful  2020-09-01 23:59:36  2020-09-01 23:37:52

<b>What is the % of sessions with an intention to book a scooter?

In [3]:
data['event_name'].unique() #To find the Unique Value of event_name and determine the values of intention to book

array(['Out-Of-Business Zone View Shown', 'Application Opened',
       'Ride Done - Successful', 'Ride Done - Unsuccessful',
       'Reduced Speed Snackbar Shown - Out Of Business Zone',
       'Vehicle Selected - User', 'Vehicle Selected - Auto',
       'Qr Scanner Tapped', 'Qr Scanner - Scanned Succeeded',
       'Ride Started - Pre-Ride', 'Ride Started - Successful',
       'Application Updated', 'Scooter Selected - User',
       'Scooter Selected - Auto',
       'Qr Scanner - Scan Failed - Vehicle Not Available',
       'Paypal Added Successfully', 'List Of Payment Shown',
       'Payment Method Chosen', 'Personal Details Added Succeeded',
       'Payment Method Screen Opened', 'Phone Number Entered',
       'Terms And Privacy Policy Accepted', 'Qr Scanner - Enter Number',
       'Ride Started - Unsuccessful', 'Add Payment Method',
       'Sidebar Opened', 'Redeem Code', 'Play Sound From Vehicle',
       'Map Legend Clicked', 'Rating Screen Closed - Maybe Later',
       'Payment De

In [4]:
intention_to_book_events = ['Application Opened','Vehicle Selected - User', 'Vehicle Selected - Auto', 
                            'Scooter Selected - User', 'Scooter Selected - Auto','Location Services Enabled']  
                            #Determining the intention to book values

intention_to_book_sessions = len(data[data['event_name'].isin(intention_to_book_events)]) / len(data)

'''Calculation to find the Intention to book percentage'''

percentage_intention_to_book = intention_to_book_sessions * 100

print(f'Percentage of sessions with intention to book: {percentage_intention_to_book:.2f}%')

Percentage of sessions with intention to book: 45.31%


<b>What is the % of sessions with a successful booking?

In [5]:
#Determining the Session with Successful Booking

successful_booking_events = ['Ride Done - Successful',
                                'Qr Scanner - Scanned Succeeded',
                                'Ride Started - Successful',
                                'Payment Method Chosen',
                                'Rating Screen Closed - Maybe Later',
                                'Payment Details Screen Opened',
                                'Rating Screen Closed - Not Interested',
                                'Ride Done - Not Charged',
                                "Credit Card Details Accepted"]

successful_book_sessions = len(data[data['event_name'].isin(successful_booking_events)]) / len(data)

#Calculation to sessions with a successful booking percentage

percentage_successful_book_sessions = successful_book_sessions * 100

print(f'Percentage of sessions with a successful booking: {percentage_successful_book_sessions:.2f}%')

Percentage of sessions with a successful booking: 18.58%


<b>What determines, according to you, an intention to book?

Based on the unique event names I determine 

1. Vehicle Selected - User

2. Vehicle Selected - Auto

3. Scooter Selected - User

4. Scooter Selected - Auto

as Intention to book events because all these event appears when a user try selecting a vehicle or scooter and 

these all are the the key step to for a booking process. Therefore I consideres the above events name as Intention to Book

<b>What are factors, according to you, determining a successful booking?

Based on the unique event names I determine

1.Qr Scanner - Scanned Succeeded

2.Ride Started - Successful

3.Payment Method Chosen

4.Rating Screen Closed - Maybe Later

5.Payment Details Screen Opened

5.Rating Screen Closed - Not Interested

6.Ride Done - Not Charged

7.Credit Card Details Accepted

as successful booking events because all these event appears when a user already booked a vehicle or scooter or after the
successful ride which takes place after successful booking. Therefore I consideres the above events name as Successful booking.

However a successful booking may vary based on the business logic and the dataset's context. It's essential to understand the 
business rules and the dataset's documentation to determine the criteria for a successful booking accurately.

<b> Do you have any hypotheses after exploring this data? If yes, please add them. 

<b>User Engagement:<b> 
    
    The relatively high percentage (45.31%) of sessions with an intention to book (based on specific events like selecting a         vehicle or scooter) suggests that users are actively engaging with the booking process. This might indicate that the             platform or service is user-friendly and encourages users to consider making a booking.

<b>Success Rates:<b> 
    
    The percentage (18.58%) of sessions with a successful booking (based on events like completing a ride or making a successful     payment) is lower than the intention-to-book rate. This could say that while users express an initial intention to book, not     all of them follow and complete a booking successfully. 
                
<b>Feedback Engagement:<b> 
    
    The engagement with feedback-related events, such as "Rating Screen Closed - Maybe Later," suggests that users are willing       to provide feedback but may choose to do so at a later time. This could indicate that users value the opportunity to provide     feedback and want to do so on their terms.
            
<b>Payment Method Adoption:<b> 
    
    The events related to payment methods ("Payment Method Chosen," "Payment Method Screen Opened," "Add Payment Method")           suggest that users are actively managing their payment methods. A hypothesis could be that offering a variety of payment         options or promotions related to payment methods might improve conversion rates.

<b>Location Services:<b> 
    
    he events related to location services ("Location Services Dialogue Opened" and "Location Services Enabled") indicate that       users are interacting with location-related features. A hypothesis could be that personalized location-based recommendations     or features could enhance the user experience.